ntge-core/src/strings.rs

```diff
@@ -11,6 +11,9 @@
 use std::ffi::{CStr, CString};
 use std::os::raw::c_char;
 
+/// # Safety
+///
+/// Left for future explaining
```

Comment:
Ref: https://github.com/fluffyemily/cross-platform-rust/blob/master/complex/cargo/ffi-utils/src/lib.rs

Reply:
This warning was addressed by clippy. We should actually add some explanation here for the safety.

Reply:
It's not truly safe. CStr is just wrapper. And here just return empty String when any error occurs. 

The safety is guarded by us. a.k.a do not misuse it.